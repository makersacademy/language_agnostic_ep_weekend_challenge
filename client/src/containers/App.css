@import url('https://fonts.googleapis.com/css?family=Aldrich|Cabin+Sketch|Julius+Sans+One|Special+Elite|Ubuntu+Mono|Black+Han+Sans');

h1 {
font-family: 'Cabin Sketch', cursive;
}

h2 {
font-family: 'Special Elite', cursive;
}
.App {
  text-align: center;
}

div {
  font-family: 'Aldrich', sans-serif
}

.date {
  font-size:10px
}



  .newButton {
    font-family: 'Cabin Sketch', cursive;
    background-color: #4CAF50; /* Green */
    border: none;
    color: white;
    padding: 16px 32px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    margin: 4px 2px;
    -webkit-transition-duration: 0.4s; /* Safari */
    transition-duration: 0.4s;
    cursor: pointer;

      color: black;
      border: 2px solid #e7e7e7;
      border-radius:5%
  }

  .newButton:hover {background-color: #e7e7e7;}



  .loginButton {
    font-family: 'Cabin Sketch', cursive;
    background-color: #4CAF50; /* Green */
    border: none;
    color: white;
    padding: 16px 32px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    margin: 4px 2px;
    -webkit-transition-duration: 0.4s; /* Safari */
    transition-duration: 0.4s;
    cursor: pointer;

      color: black;
      border: 2px solid #e7e7e7;
      border-radius:5%
  }

  .loginButton:hover {background-color: #e7e7e7;}

  .headerButtons{
    font-family: 'Cabin Sketch', cursive;
    background-color: #4CAF50; /* Green */
    border: none;
    color: white;
    padding: 16px 32px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    margin: 4px 2px;
    -webkit-transition-duration: 0.4s; /* Safari */
    transition-duration: 0.4s;
    cursor: pointer;

      color: black;
      border: 2px solid #e7e7e7;
      border-radius:5%
  }

  .headerButtons:hover {background-color: #e7e7e7;}
/* .fonts {
  font-family: 'Aldrich', sans-serif;
font-family: 'Cabin Sketch', cursive;
font-family: 'Special Elite', cursive;
font-family: 'Julius Sans One', sans-serif;
font-family: 'Ubuntu Mono', monospace;
} */
/* .App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
} */

.App-header {
  background-color: #822b2b;
  /* min-height: 100vh; */
  height:200px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

/* .App-link {
  color: #61dafb;
} */

/* @keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
} */

.gallery {
  display: flex;
  flex-wrap: wrap;
  align-content: flex-start;
  text-align: center;
  margin:auto;
}
.centered {
  text-alight:center;
}
.gallery_item {
  width: 30%;
  height: 250px;
  border: 1px solid black;
}
.gallery_image {
  width: 100%;
  height: 100%;
   cursor: pointer;
}

div.gallery_item {
    border: 1px solid #ccc;
}

div.gallery_item:hover {
    border: 1px solid #777;
}

div.gallery_item img {
    width: 100%;
    height: auto;
}

div.desc {
    padding: 15px;
    text-align: center;
}

* {
    box-sizing: border-box;
}

.responsive {
    padding: 0 6px;
    float: left;
    width: 24.99999%;
}

@media only screen and (max-width: 700px) {
    .responsive {
        width: 49.99999%;
        margin: 6px 0;
    }
}

@media only screen and (max-width: 500px) {
    .responsive {
        width: 100%;
    }
}

.clearfix:after {
    content: "";
    display: table;
    clear: both;
}
