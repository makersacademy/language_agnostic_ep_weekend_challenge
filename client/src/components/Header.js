import React, { Component } from 'react';
import '../containers/App.css'

class Header extends Component {

constructor(props) {
  super(props)
console.log(this.props)
}

componentWillReceiveProps({someProp}) {
  console.log("PROPS")
  this.setState(this.state)
  console.log(this.props.loggedIn)
}

  render() {
    return (
      // {this.props.togglenewpost}
      <div className="App-header">
      <h1>!Instagram</h1>
      <div>

      { this.props.loggedIn && <button name="newPost" className="headerButtons" id="newPost" onClick={this.props.toggleNewPost}>New Post</button> }
      { !this.props.loggedIn && <button name="Login" className="headerButtons" id="Login" onClick={this.props.toggleLogin}>Login or Register</button>}User: {this.props.user}
      { this.props.loggedIn && <button name="logout" className="headerButtons" id="logout" onClick={this.props.toggleLogout}>Logout</button> }
      </div>
      </div>
    )
  }
}

export default Header
