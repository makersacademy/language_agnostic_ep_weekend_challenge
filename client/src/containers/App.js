import React, { Component } from 'react';
import FileUpload from '../components/FileUpload'
import Gallery from './Gallery'
import Login from './Login'
import Header from '../components/Header'

import './App.css';

class App extends Component {

  constructor(props){
    super(props)
    var loggedIn
    console.log(sessionStorage.getItem('myData'))
    if (sessionStorage.getItem('myData')=== null) {
      loggedIn = false
    } else {
      loggedIn = true
    }
    this.state = {
      newposthidden:true,
      loginhidden:true,
      user: sessionStorage.getItem('myData'),
      loggedIn: loggedIn
    }

    this.toggleNewPost = this.toggleNewPost.bind(this);
    this.toggleLogin = this.toggleLogin.bind(this);
    this.toggleLogout = this.toggleLogout.bind(this);
    this.setUser = this.setUser.bind(this);
  }

  toggleNewPost() {

    console.log("toggle?")
    this.setState({newposthidden:!this.state.newposthidden})
  }
  toggleLogin() {
    // alert('test')
    console.log("toggle?")
    this.setState({loginhidden:!this.state.loginhidden})
  }
  toggleLogout() {
    console.log("toggle?")
    sessionStorage.clear()
    this.setState({user:'', loggedIn:false})

  }

  setUser(user) {
    sessionStorage.setItem('myData', user);
    this.setState({user: sessionStorage.getItem('myData'), loggedIn:true})
    window.location.reload()
  }

  render() {
    return (
      <div className="App">
        <Header toggleNewPost={this.toggleNewPost} toggleLogin={this.toggleLogin} toggleLogout={this.toggleLogout} loggedIn={this.state.loggedIn} user={this.state.user}/>
        {!this.state.newposthidden && <FileUpload />}
        {!this.state.loginhidden && <Login setUser={this.setUser}/>}

        <Gallery/>


      </div>
    );
  }
}

export default App;
