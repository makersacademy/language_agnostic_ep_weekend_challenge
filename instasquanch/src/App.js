import React, { Component } from 'react';
import Header from './containers/header'
import Body from './containers/body'

class App extends Component {
  constructor(){
    super()
    this.state = {
      images: [{src:'https://f4.bcbits.com/img/0010067948_20.jpg'},
      {src:'https://i.ytimg.com/vi/0vfLj6kaSFY/maxresdefault.jpg'},
      {src:'https://i.redd.it/tfp6ryte6fsy.jpg'},
      {src:'https://i.imgflip.com/tjezd.jpg'}]
    }
  }

  componentDidMount(){

  }
  render() {
    return (
      <div className='headerDiv'>
        <Header />
        <div className='imageList'>
        {this.state.images.map ((image, index) => {
          return <div className ='img' key={'lisi' + index}> <Body src={image.src} key={index}/></div>
        })
      }
    </div>
      </div>
    );
  }
}

export default App;
