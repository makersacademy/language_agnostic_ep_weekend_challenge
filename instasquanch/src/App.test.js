import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Header from './containers/header.js'
import Enzyme, { mount, shallow, render } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('LandingPage', () => {
  let props
  let mountedAppPage
  const appPage = () => {
    mountedAppPage = mount(<App {...props}/>)
    return mountedAppPage
  }

  beforeEach(() => {
    props = {
      images: undefined
    }
    mountedAppPage = undefined
  })

  it('always renders a div', () => {
    const divs = appPage().find('div')
    expect(divs.length).toBeGreaterThan(0)
  })

  it('should render correctly in "debug" mode', () => {
    const component = shallow(<App debug />);

    expect(component).toMatchSnapshot();
  });
})
