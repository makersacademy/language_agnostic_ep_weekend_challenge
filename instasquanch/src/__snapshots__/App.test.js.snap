// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <Body
          src={
            Array [
              "https://f4.bcbits.com/img/0010067948_20.jpg",
              "https://i.ytimg.com/vi/0vfLj6kaSFY/maxresdefault.jpg",
              "https://i.redd.it/tfp6ryte6fsy.jpg",
              "https://vignette.wikia.nocookie.net/rickandmorty/images/1/1c/S2e10_squanchy_not_ricks_bff.png/revision/latest?cb=20160923050635",
            ]
          }
        />,
      ],
      "className": "headerDiv",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "src": Array [
            "https://f4.bcbits.com/img/0010067948_20.jpg",
            "https://i.ytimg.com/vi/0vfLj6kaSFY/maxresdefault.jpg",
            "https://i.redd.it/tfp6ryte6fsy.jpg",
            "https://vignette.wikia.nocookie.net/rickandmorty/images/1/1c/S2e10_squanchy_not_ricks_bff.png/revision/latest?cb=20160923050635",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <Body
            src={
              Array [
                "https://f4.bcbits.com/img/0010067948_20.jpg",
                "https://i.ytimg.com/vi/0vfLj6kaSFY/maxresdefault.jpg",
                "https://i.redd.it/tfp6ryte6fsy.jpg",
                "https://vignette.wikia.nocookie.net/rickandmorty/images/1/1c/S2e10_squanchy_not_ricks_bff.png/revision/latest?cb=20160923050635",
              ]
            }
          />,
        ],
        "className": "headerDiv",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "src": Array [
              "https://f4.bcbits.com/img/0010067948_20.jpg",
              "https://i.ytimg.com/vi/0vfLj6kaSFY/maxresdefault.jpg",
              "https://i.redd.it/tfp6ryte6fsy.jpg",
              "https://vignette.wikia.nocookie.net/rickandmorty/images/1/1c/S2e10_squanchy_not_ricks_bff.png/revision/latest?cb=20160923050635",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
